Issue 92

LightAllWorld opened the pull request:
This PR fixes issue #92 :

It adds an setter `set_tesseract_path` about tesseract path.

JaMe76 commented on deepdoctection/utils/file_utils.py:
```diff
@@ -249,8 +249,23 @@ def get_detectron2_requirement() -> Requirement:
+_TESS_PATH = "tesseract"
+
+def set_tesseract_path(tesseract_path: str="tesseract"):
+    global _TESS_AVAILABLE
```
Please add a short docstring, e.g. what path to add for installation with Anaconda

JaMe76 commented on deepdoctection/utils/file_utils.py:
```diff
@@ -249,8 +249,23 @@ def get_detectron2_requirement() -> Requirement:
 _TESS_AVAILABLE = which("tesseract") is not None
+_TESS_PATH = "tesseract"
+
+def set_tesseract_path(tesseract_path: str="tesseract"):
```
function will need a typing return statement:

```
set_tesseract_path(tesseract_path: str="tesseract") -> None:
```

JaMe76 reviewed (changes requested):
Thanks for your PR!

1.) To make the PATH changes to tesseract effective in predictions _TESS_PATH needs to be imported to `extern.tessocr` and _TESS_PATH must replace "tesseract" in line 96. 

```
cmd_args += ("tesseract", input_file_name, output_file_name_base, "-l", lang)
```

2.) Can you add `set_tesseract_path` to `deepdoctection.__init__.py`  in _IMPORT_STRUCTURE dict in line 265? All modules are imported in a lazy manner, so not adding this function here would result in import error when invoked.   

3.) I've added some small remarks in your changes. 

Once merged, I would convert the original issue into the discussion section so users from Anaconda installs do not need to dig into closed issues. 

Thanks

LightAllWorld commented:
Thanks for your comments.

1. Added doc strings about `set_tesserect_path` and `_TESS_PATH'.

2. Replaced extern.tessocr's "tesserect" in line 96.

3. Add `set_tesseract_path' to `deepdoctection.__init__.py` in 265 line.

JaMe76 reviewed (approved):
Thanks for your updates. 

Will be merged.
